app/src/org/commcare/android/models/AndroidSessionWrapper.java

```diff
@@ -299,9 +288,12 @@ public String getTitle() {
         
         //TODO: This manipulates the state of the session. We should instead grab and make a copy of the frame, and make a new session to 
         //investigate this.
-        
-        //Walk backwards until we find something with a long detail
-        while(session.getFrame().getSteps().size() > 0 && (session.getNeededData() != SessionFrame.STATE_DATUM_VAL || session.getNeededDatum().getLongDetail() == null)) {
+
+        // Walk backwards until we find something with a long detail
+        while (session.getFrame().getSteps().size() > 0 &&
+                (!SessionFrame.STATE_DATUM_VAL.equals(session.getNeededData()) ||
+                        !(session.getNeededDatum() != null &&
+                                session.getNeededDatum().getLongDetail() != null))) {
```

Comment:
Is there a specific reason? If session.getNeededData() is returning a datum but getNeededDatum() is null that feels like a pretty serious issue. I might suggest failing inside getNeededData structurally to make the nature of the error more clear rather than bubbling up to here.